### PR TITLE
Checkboxes

### PR DESCRIPTION
**Pull request by AlehandroL**

### CODE REVIEW

File: app/controllers/resource_comments_controller.rb [hunk 1]
```diff
@@ -7,6 +7,12 @@ def create
       user: current_user,
       resource_id:
     )
-    redirect_to(Resource.find(resource_id))
+    redirect_to_resource(resource_id)
+  end
+
+  private
+
+  def redirect_to_resource(resource_id)
```
Comment (ristofer) [inline app/controllers/resource_comments_controller.rb]: I am against this method.  Why could be worth writing it?

File: app/controllers/resource_evaluations_controller.rb [hunk 2]
```diff
@@ -7,6 +7,12 @@ def create
                                                  resource_id:)
     resource_evaluation.evaluation = evaluation
     resource_evaluation.save if resource_evaluation.has_changes_to_save?
-    redirect_to(Resource.find(resource_id))
+    redirect_to_resource(resource_id)
+  end
+
+  private
+
+  def redirect_to_resource(resource_id)
```
Comment (ristofer) [inline app/controllers/resource_evaluations_controller.rb]: I think this is another reason to don't have this method. Is defined on two different files.

File: app/services/learning_units/display_user_learning_unit.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      @curriculum = curriculum
+      @user = current_user
+    end
+
+    def user_learning_units
+      @curriculum.learning_units.map do |learning_unit|
+        {
+          learning_unit:,
+          user: @user,
+          completed_by_user: CompletedLearningUnit.find_by(
```
Comment (ristofer) [inline app/services/learning_units/display_user_learning_unit.rb:13]: Could be better to make a special query for this? For each learning_unit there is a query searching the completed learning unit.

File: spec/requests/completed_learning_units_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        )
+      end
+
+      it 'creates a new record in CompletedLearningUnits' do
+        expect { perform }.to change(CompletedLearningUnit, :count).by(1)
+      end
+
+      it 'redirects to curriculum page' do
+        expect(perform).to redirect_to(
+          root_path
```
Comment (ristofer) [inline spec/requests/completed_learning_units_controller_spec.rb:27]: I would change root_path to curriculum_path(Curriculum)

File: app/controllers/resource_comments_controller.rb (hunk 1 above)
Comment (AlehandroL) [reply app/controllers/resource_comments_controller.rb]: It's not.

File: spec/requests/completed_learning_units_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        )
+      end
+
+      it 'shows a new button with a completed message' do
```
Comment (ristofer) [inline spec/requests/completed_learning_units_controller_spec.rb:31]: I would add a better description to understand the two redirects. Like "following two redirects to the ____ and the ___ it shows a new button with a completed message"

File: spec/requests/completed_learning_units_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      )
+    end
+
+    it 'deletes a record in CompletedLearningUnits' do
+      expect { perform }.to change(CompletedLearningUnit, :count).by(-1)
+    end
+
+    it 'redirects to curriculum page' do
+      expect(perform).to redirect_to(
+        root_path
```
Comment (ristofer) [inline spec/requests/completed_learning_units_controller_spec.rb:70]: Same thing as before. Better to specify the curriculum_path.

File: spec/requests/resources_controller_spec.rb [hunk 7]
```diff
@@ -29,7 +29,7 @@
     end
 
     def perform
-      get resource_path(resource)
+      get resource_path(resource.id)
```
Comment (ristofer) [inline spec/requests/resources_controller_spec.rb:32]: I don't know if this change was necessary. It was?

**Review by ristofer — APPROVED** [review]

**Pull request merged**